File: 2021/src/test/java/de/jfriemel/aoc/days/Day03Test.java
package de.jfriemel.aoc.days;

import org.junit.Test;

import java.util.Arrays;
import java.util.List;

import static org.junit.Assert.assertEquals;

public class Day03Test {

    private static final String INPUT_STRING = """
            00100
            11110
            10110
            10111
            10101
            01111
            00111
            11100
            10000
            11001
            00010
            01010""";

    private static final List<String> INPUT = Arrays.asList(INPUT_STRING.split("\n"));

    @Test
    public void testPart1() {
        String result = new Day03().part1(INPUT);
        assertEquals("198", result);
    }

    @Test
    public void testPart2() {
        String result = new Day03().part2(INPUT);
        assertEquals("230", result);
    }

}

File: 2023/src/test/kotlin/de/jfriemel/aoc/days/Day03Test.kt
package de.jfriemel.aoc.days

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class Day03Test {
    private val input = """
        467..114..
        ...*......
        ..35..633.
        ......#...
        617*......
        .....+.58.
        ..592.....
        ......755.
        ...${'$'}.*....
        .664.598..
    """.trimIndent().lines()

    @Test
    fun testPart1() {
        val result = Day03.part1(input)
        Assertions.assertEquals("4361", result)
    }

    @Test
    fun testPart2() {
        val result = Day03.part2(input)
        Assertions.assertEquals("467835", result)
    }
}
